warehouse/jobs/runner.go

```diff
@@ -107,11 +109,11 @@ func (a *AsyncJobWh) addJobsToDB(payload *AsyncJobPayload) (int64, error) {
 	return jobId, nil
 }
 
-// InitAsyncJobRunner Async Job runner's main job is to
+// Start Async Job runner's main job is to
 // 1. Scan the database for entries into wh_async_jobs
 // 2. Publish data to pg_notifier queue
 // 3. Move any executing jobs to waiting
-func (a *AsyncJobWh) InitAsyncJobRunner() error {
+func (a *AsyncJobWh) Start() error {
 	// Start the asyncJobRunner
 	a.logger.Info("[WH-Jobs]: Initializing async job runner")
 	g, ctx := errgroup.WithContext(a.context)
```

Comment:
> Do we need the group in the first place? Since it is only really using a single routine, shouldn't we just remove it and block at the end with a return a.startAsyncJobRunner(ctx)?

> It would be great if we can have it accept a context as well. Then once the context is canceled Run can return after gracefully terminating all the routines that it started.

I will try to take these up in a separate PR. I have updated the [linear ticket](https://linear.app/rudderstack/issue/PIPE-168/jobs-cleanup) regarding this.